### PR TITLE
[Big data] Counterfactuals - part 1

### PR DESCRIPTION
**Pull request by vinuthakaranth**
Signed-off-by: vinutha karanth <vinutha.karanth@gmail.com>

## Description

This PR contains all the other changes needed for counterfactuals other than changes in libs/counterfactuals folder.
Next part will contain all changes in libs/counterfactuals folder.

## Checklist

- [ ] I have added screenshots above for all UI changes.
- [ ] I have added e2e tests for all UI changes.
- [ ] Documentation was updated if it was needed.

### CODE REVIEW

File: apps/widget/src/app/ModelAssessment.tsx [hunk 1]
```diff
@@ -160,6 +163,24 @@ export class ModelAssessment extends React.Component<IModelAssessmentProps> {
           abortSignal
         );
       };
+      callBack.requestBubblePlotData = async (
+        data: any
+      ): Promise<IHighchartBubbleSDKData> => {
+        return callFlaskService(
+          this.props.config,
+          data,
+          "/dataset_analysis_bubble_chart_plot"
```
[the remark this answers is not in the document]
Comment (vinuthakaranth) [reply apps/widget/src/app/ModelAssessment.tsx:172]: thanks Roman, created task to follow up on this as this needs sdk change.

File: apps/dashboard/src/model-assessment/__mock_data__/adultCensus.ts [hunk 2]
```diff
@@ -12047,6 +12047,7 @@ export const adultCounterfactualData: ICounterfactualData = {
     "native-country",
     "income"
   ],
+  id: "abc",
```
[the remark this answers is not in the document]
Comment (vinuthakaranth) [reply apps/dashboard/src/model-assessment/__mock_data__/adultCensus.ts:12050]: Id is counterfactualsData id. This is sent by sdk. Id along with absolute index gives us the local counterfactual data. Tagging @gaugup to add more context on this.

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/ModelAssessmentDashboardProps.ts [hunk 3]
```diff
@@ -110,6 +111,14 @@ export interface IModelAssessmentDashboardProps
...
   ) => Promise<any>;
   requestExp?: (index: number, abortSignal: AbortSignal) => Promise<any[]>;
+  requestBubblePlotData?: (
+    request: any[],
+    abortSignal: AbortSignal
+  ) => Promise<IHighchartBubbleSDKData>;
+  requestLocalCounterfactuals?: (
+    request: any[],
+    abortSignal: AbortSignal
+  ) => Promise<any>;
```
[the remark this answers is not in the document]
Comment (vinuthakaranth) [reply libs/model-assessment/src/lib/ModelAssessmentDashboard/ModelAssessmentDashboardProps.ts:121]: Thanks Roman for catching this, I will keep this in mind and use the counterfactuals interface in my coming PR after testing.

**Review by tongyu-microsoft — APPROVED** [review]

File: libs/core-ui/src/lib/Interfaces/IHighchartBubbleData.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  id: string;
+  name: string;
+  size: number;
+  x: any;
+  y: any;
+  index_series: number[];
+  x_series: any[];
+  y_series: any[];
+}
+
```
[the remark this answers is not in the document]
Comment (vinuthakaranth) [reply libs/core-ui/src/lib/Interfaces/IHighchartBubbleData.ts:14]: added comment

File: apps/dashboard/src/model-assessment/__mock_data__/adultCensus.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (gaugup) [reply apps/dashboard/src/model-assessment/__mock_data__/adultCensus.ts:12050]: ID is populated by SDK to identify which counterfactual configuration should respond to the UI flask server call.

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/ModelAssessmentDashboardProps.ts [hunk 5]
```diff
@@ -110,6 +111,14 @@ export interface IModelAssessmentDashboardProps
     abortSignal: AbortSignal
   ) => Promise<any>;
   requestExp?: (index: number, abortSignal: AbortSignal) => Promise<any[]>;
+  requestBubblePlotData?: (
```
Comment (imatiach-msft) [inline libs/model-assessment/src/lib/ModelAssessmentDashboard/ModelAssessmentDashboardProps.ts:114]: I feel like we have a lot of these request methods.  Any ideas on ways we could organize them better?

**Review by imatiach-msft — APPROVED** [review]

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/ModelAssessmentDashboardProps.ts (hunk 5 above)
Comment (vinuthakaranth) [reply libs/model-assessment/src/lib/ModelAssessmentDashboard/ModelAssessmentDashboardProps.ts:114]: Thanks Ilya for the comment. This is great idea! We can have a meeting with team to discuss about organizing these as we will have more requests methods to come in future.

File: libs/interpret/src/lib/MLIDashboard/Controls/WhatIfTab/WhatIfTab.tsx [hunk 6]
```diff
@@ -403,6 +404,7 @@ export class WhatIfTab extends React.PureComponent<
                     canDither={
                       this.state.chartProps.chartType === ChartTypes.Scatter
                     }
+                    allowTreatAsCategorical
```
Comment (gaugup) [inline libs/interpret/src/lib/MLIDashboard/Controls/WhatIfTab/WhatIfTab.tsx:407]: What does the new variable ''allowTreatAsCategorical" do? Looks like it is used in a lot of places.

**Review by gaugup — COMMENTED** [review]
LGTM. I have one question in the PR if you could help clarify/

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/ModelAssessmentDashboardProps.ts (hunk 5 above)
Comment (gaugup) [reply libs/model-assessment/src/lib/ModelAssessmentDashboard/ModelAssessmentDashboardProps.ts:114]: @imatiach-msft  we have an internal task in engineering excellence work to organize the new end points in some hierarchy.

**Review by gaugup — APPROVED** [review]

File: libs/interpret/src/lib/MLIDashboard/Controls/WhatIfTab/WhatIfTab.tsx (hunk 6 above)
Comment (vinuthakaranth) [reply libs/interpret/src/lib/MLIDashboard/Controls/WhatIfTab/WhatIfTab.tsx:407]: Discussed offline. allowTreatAsCategorical is new param to allow that toggle or not. As we decided for big data we dont allow treat as cat', I need to add an extra param to handle this case. For big data allowTreatAsCategorical is sent as false. allowTreatAsCategorical is true we show toggle .

**Pull request merged**